include/deal.II/particles/particle_accessor.h

```diff
@@ -593,14 +599,16 @@ namespace Particles
                 break;
               }
 
-            if ((*particles)[active_cell_index].size() > 0)
+            if ((*particles)[active_cell_index].size() > 0 &&
+                cell->is_locally_owned() == initial_cell_is_owned)
               {
                 particle_index_within_cell =
                   (*particles)[active_cell_index].size() - 1;
                 break;
               }
           }
-        while ((*particles)[active_cell_index].size() == 0);
+        while ((*particles)[active_cell_index].size() == 0 &&
```

Comment:
Shouldnt this be the following:

```suggestion
        while ((*particles)[active_cell_index].size() == 0 ||
```

Reply:
Yup that's true. Fixed.